shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/viewModel/FavoritesViewModel.kt

```diff
@@ -0,0 +1,186 @@
+                    stop.childStopIds.filter { globalData?.stops?.containsKey(it) ?: false } +
+                        stop.id
+                }
+            }
+        val schedules = getSchedules(stopIds, "FavoritesViewModel.getSchedules")
+        val predictions = subscribeToPredictions(stopIds, active)
+
+        LaunchedEffect(Unit) { favorites = favoritesUsecases.getRouteStopDirectionFavorites() }
+
+        LaunchedEffect(Unit) {
+            events.collect { event ->
+                when (event) {
+                    Event.ReloadFavorites ->
+                        favorites = favoritesUsecases.getRouteStopDirectionFavorites()
+                    is Event.SetActive -> {
+                        active = event.active
+                        event.isReturningFromBackground?.let { isReturningFromBackground = it }
+                    }
+                    is Event.SetAlerts -> alerts = event.alerts
+                    is Event.SetLocation -> location = event.location
+                    is Event.SetNow -> now = event.now
+                }
+            }
+        }
+
+        LaunchedEffect(predictions) {
+            if (predictions != null) {
+                isReturningFromBackground = false
+            }
+        }
+
+        LaunchedEffect(
+            stopIds,
+            globalData,
+            location,
+            schedules,
+            predictions,
+            alerts,
+            now,
+            favorites,
+        ) {
+            if (stopIds == null || globalData == null || location == null) {
+                routeCardData = null
+            } else if (stopIds.isEmpty()) {
+                routeCardData = emptyList()
+            } else {
+                val loadedRouteCardData =
+                    RouteCardData.routeCardsForStopList(
+                        stopIds,
+                        globalData,
+                        location,
+                        schedules,
+                        predictions,
+                        alerts,
+                        now,
+                        emptySet(),
+                        RouteCardData.Context.Favorites,
+                    )
+                routeCardData = filterRouteAndDirection(loadedRouteCardData, globalData, favorites)
+            }
+        }
+
+        LaunchedEffect(stopIds, globalData, favorites) {
+            if (stopIds == null || globalData == null) {
+                staticRouteCardData = null
+            } else if (stopIds.isEmpty()) {
+                staticRouteCardData = emptyList()
+            } else {
+                val loadedRouteCardData =
+                    RouteCardData.routeCardsForStaticStopList(
+                        stopIds,
+                        globalData,
+                        RouteCardData.Context.Favorites,
+                    )
+                staticRouteCardData =
+                    filterRouteAndDirection(loadedRouteCardData, globalData, favorites)
```

Comment:
Strictly speaking, the static route card data isn’t required until we implement favorites deletion, but that’s basically the next ticket, and it’ll be a little bit annoying to add in later instead (since the static data loads before the realtime data in the test).